### PR TITLE
X3D serializer: Material RGB colors

### LINKED ISSUE #1231: x3d serializer: Material colors should be RGB, not RGBA
## Expected Behavior

x3d serializer should generate

```xml
<Appearance>
        <Material diffuseColor="0 1 0" emissiveColor="0 1 0" transparency="0.2"/>
</Appearance
```

## Actual Behavior

```xml
<Appearance>
        <Material diffuseColor="0 1 0 0.8" emissiveColor="0 1 0 0.8"/>
</Appearance>
```

X3D Material uses RGB format with a separate transparency field.

## Steps to Reproduce the Problem

  1. load basic colors example on https://www.openjscad.xyz/
  2. export to X3D file
  3. find Material node

## Specifications

  - Version: latest
  - Platform: web
  - Environment: browser

X3D: Material SFColor fields, SFColor is RGB
https://www.web3d.org/documents/specifications/19775-1/V3.3/Part01/components/shape.html#Material
https://www.web3d.org/documents/specifications/19775-1/V3.3/Part01/fieldsDef.html#SFColorAndMFColor

### PR DESCRIPTION
**Pull request by andreasplesch**
### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [x] Does your submission pass tests?

fixes #1231

some additional cleanup as suggested by standardx.

### CODE REVIEW

**Comment by andreasplesch** [comment]
I will be glad to address any review comments.

**Review by hrgdavor — APPROVED** [review]
looks good :)

File: packages/io/x3d-serializer/src/index.js [hunk 1]
```diff
@@ -177,7 +177,7 @@ const convertAppearance = (object, options) => {
   const colorRGB = object.color.slice(0, 3)
   const diffuseColor = colorRGB.join(' ')
   const emissiveColor = colorRGB.join(' ')
-  const transparency = object.color[3]
+  const transparency = 1.0 - object.color[3]
```
Comment (andreasplesch) [inline packages/io/x3d-serializer/src/index.js:180]: Sorry for the late change. I had never pushed this correction.

**Review by z3dev — APPROVED** [review]
@andreasplesch thanks again for these changes.

pits nice to have someone else looking at this code, and making improvements.

**Comment by andreasplesch** [comment]
Thanks, no problem.

**Pull request merged**